### PR TITLE
WFS client: remove logic of InvalidParameterValue for outputformat

### PR DESCRIPTION
**Pull request by jampukka**
Many servers just ignore the invalid parameter and respond with GML anyways when requesting GeoJSON. Remove the code that checks if GeoJSON parsing failed because it was an OWSException stating that the specified value for outputformat parameter was invalid.

Also some refactoring.

### CODE REVIEW

File: service-wfs-client/src/main/java/org/oskari/service/wfs/client/OskariWFSClient.java [hunk 1]
```diff
@@ -61,17 +65,63 @@ protected static String getBBOX(ReferencedEnvelope bbox) {
...
+            responseHeaders = OskariWFSClient.readResponseTo(endPoint, user, pass, query, baos);
+            response = baos.toByteArray();
+            // TODO: Select parsing algorithm based on response headers (Content-Type)
+            fc = parseGeoJSON(response, crs, url);
+            if (fc != null) {
+                return fc;
+            }
+            // Try to parse the same response as GML
+            fc = parseGML(response, crs, url, user, pass, gmlDecoder);
+            if (fc != null) {
```
Comment (ZakarFin) [inline service-wfs-client/src/main/java/org/oskari/service/wfs/client/OskariWFSClient.java]: Add:

LOG.warn("Requested JSON but got GML. Possibly misconfigured service for", url); or something similar.

File: service-wfs-client/src/main/java/org/oskari/service/wfs/client/OskariWFSClient.java [hunk 2]
```diff
@@ -61,17 +65,63 @@ protected static String getBBOX(ReferencedEnvelope bbox) {
...
+                return fc;
+            }
+            // Okay I guess it wasn't a GML FeatureCollection either - move on
```
Comment (ZakarFin) [inline service-wfs-client/src/main/java/org/oskari/service/wfs/client/OskariWFSClient.java]: Add:

LOG.warn("Requested JSON but didn't get a parseable result. Possibly misconfigured service for", url); or something similar.

Comment (jampukka) [reply service-wfs-client/src/main/java/org/oskari/service/wfs/client/OskariWFSClient.java]: Added

File: service-wfs-client/src/main/java/org/oskari/service/wfs/client/OskariWFSClient.java (hunk 1 above)
Comment (jampukka) [reply service-wfs-client/src/main/java/org/oskari/service/wfs/client/OskariWFSClient.java]: Added

**Pull request merged**